content/docs/rfcs/38/README.md

```diff
@@ -0,0 +1,76 @@
+
+# Theory / Semantics
+
+As explained in [17/WAKU-RLN-RELAY](https://rfc.vac.dev/spec/17/), all peers who want to publish the message in the spam protected peer to peer network MUST register for membership to the RLN group. This is done via the `register` function of the RLN membership contract. If a registered member of the RLN group spams messages in the network then its secret key get exposed which can be used to slash its membership from the group. This is done by calling the `withdraw` function of the RLN membership contract, which removes the member from the merkle tree thereby revoking its rights to send any messages.
+
+# Wire Format Specification / Syntax
+
+### Constructor
+
+* `constructor(uint256 membershipDeposit, uint256 depth, address _poseidonHasher) public`
+    * `membershipDeposit` sets the amount of ETH that must be deposited to `register` to the RLN group.
+    * `depth` is the depth of the merkle tree that determines the maximum number of members to the RLN group.
+    * `_poseidonHasher` is the address of the poseidon hasher contract used in the `hash` function.
+
+### Methods
+
+* register: This function registers the `pubkey` to the list of members. For succesful registration, ETH equal to the amount specified in `membershipDeposit` MUST be paid while calling this function. This function MUST fire the `MemberRegistered` event.
+    * `function register(uint256 pubkey) external payable`
+
+
+
+* registerBatch: This function registers multiple `pubkeys` to the list of members. For succesful registration, ETH equal to the amount specified in `membershipDeposit * pubkeys.length` MUST be paid while calling this function. This function MUST fire the `MemberRegistered` event for each registered member.
+    * `function registerBatch(uint256[] calldata pubkeys) external payable`
+
+
+* withdraw: This function accepts the `secret` of the public key at index`_pubkeyIndex` to withdraw the ETH equal to the`membershipDeposit` to the `receiver` address. This function MUST remove the associated member from the list of members. This function MUST fire the `MemberWithdrawn` event.
+    * `function withdraw(uint256 secret, uint256 _pubkeyIndex, address payable receiver) external`
+
+
+* withdrawBatch: This function accepts multiple `secrets` of the public keys at the indices `pubkeyIndexes` to withdraw the ETH equal to the `membershipDeposit * secrets.length` to the `receiver` address. This function MUST remove the associated members from the list of members. This function MUST fire the `MemberWithdrawn` event for each member slashed from the group.
+    * `function withdrawBatch(uint256[] calldata secrets, uint256[] calldata pubkeyIndexes, address payable[] calldata receivers) external`
```

Comment:
Same an above, receiver should not be included to avoid MEV attacks.

Reply:
Noted.